protocol/x/clob/keeper/process_operations.go

```diff
@@ -630,17 +635,26 @@ func (k Keeper) PersistMatchDeleveragingToState(
 				err,
 			),
 		)
-	} else if !canDeleverageSubaccount {
+	} else if !shouldDeleverageAtBankruptcyPrice && !shouldDeleverageAtOraclePrice {
 		// TODO(CLOB-853): Add more verbose error logging about why deleveraging failed validation.
 		return errorsmod.Wrapf(
 			types.ErrInvalidDeleveragedSubaccount,
 			"Subaccount %+v failed deleveraging validation",
 			liquidatedSubaccountId,
 		)
+	} else if matchDeleveraging.IsFinalSettlement != shouldDeleverageAtOraclePrice {
+		// Throw error if the isFinalSettlement flag does not match the expected value. This prevents misuse or lack
+		// of use of the isFinalSettlement flag. The isFinalSettlement flag should be set to true if-and-only-if the
+		// subaccount has non-negative TNC and the market is in final settlement. Otherwise, it must be false.
+		return errorsmod.Wrapf(
+			types.ErrDeleveragingIsFinalSettlementFlagMismatch,
+			"MatchPerpetualDeleveraging %+v has isFinalSettlement flag (%v), expected (%v)",
+			matchDeleveraging,
+			matchDeleveraging.IsFinalSettlement,
+			shouldDeleverageAtOraclePrice,
+		)
 	}
```

Comment:
Suggestion: can we split these out into different if statements? IMO `else if` statements are more difficult to reason about